### PR TITLE
feat(controller): add support for CiliumNetworkPolicy generation

### PR DESCRIPTION
**Pull request by akila-i**
## Purpose
This pull request adds support for generating CiliumNetworkPolicy (CNP) resources with L7 HTTP rules for per-component network policies, in addition to the existing Kubernetes NetworkPolicy support. The network policy provider can now be configured via an environment variable, and Helm chart options have been added to make this configurable. The codebase is updated to generate the correct policy type based on the provider, and comprehensive unit tests are included for the new Cilium policy logic.

**Network Policy Provider Support:**

* Introduced a `Provider` type in `internal/networkpolicy` to select between standard Kubernetes NetworkPolicy and CiliumNetworkPolicy with L7 HTTP rules. The provider is configurable via the `NETWORK_POLICY_PROVIDER` environment variable. [[1]](diffhunk://#diff-195fe80ecf9f4af12a9ba13c2b9892ccf556b7cd73fd63334159dd6e24832b5aR9-R30) [[2]](diffhunk://#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6R57) [[3]](diffhunk://#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6R218-R227) [[4]](diffhunk://#diff-440c67a69f0874087394cbd13c783e3bfd5465ecbac9df9f2dce3cf2dbd95fbeR51-R54) [[5]](diffhunk://#diff-440c67a69f0874087394cbd13c783e3bfd5465ecbac9df9f2dce3cf2dbd95fbeL512-R524)
* Updated the `MakeComponentPolicies` function to generate either a standard NetworkPolicy or a CiliumNetworkPolicy based on the selected provider. Added logic for CNP generation, including L7/L4 port grouping and visibility rules. [[1]](diffhunk://#diff-195fe80ecf9f4af12a9ba13c2b9892ccf556b7cd73fd63334159dd6e24832b5aR40-R51) [[2]](diffhunk://#diff-195fe80ecf9f4af12a9ba13c2b9892ccf556b7cd73fd63334159dd6e24832b5aR195-R346)
* NetworkPolicyProvider is determined at policy generation time based on an annotation in the DataPlane CR.

**Testing:**

* Added comprehensive unit tests in `internal/networkpolicy/networkpolicy_test.go` to verify CiliumNetworkPolicy generation for various endpoint types, visibilities, and edge cases (such as name truncation).

## Related Issues
https://github.com/openchoreo/openchoreo/issues/3103

## Checklist
- [x] Tests added or updated (unit, integration, etc.)
- [ ] Samples updated (if applicable)
- [ ] Added `backport/<release-branch>` label if this should be backported (e.g., `backport/release-v1.0`)

### CODE REVIEW

File: internal/networkpolicy/networkpolicy.go [hunk 1]
```diff
@@ -169,6 +192,168 @@ func makeIngressRules(params ComponentPolicyParams) []any {
...
+					},
+					"matchExpressions": []any{
+						map[string]any{ // Explicitly allow from any namespace
+							"key":      KubernetesNamespaceKey,
+							"operator": "Exists",
+						},
+					},
+				},
+			},
+			"toPorts": ciliumToPorts(namespacePorts),
```
[the remark this answers is not in the document]
Comment (akila-i) [reply internal/networkpolicy/networkpolicy.go:269]: `openchoreo.dev/namespace`, `openchoreo.dev/environment`, and similar labels are applied to all resources that are being deployed to the dataplane (including pods). Refer to the podSelectors in https://github.com/openchoreo/openchoreo/blob/main/internal/controller/releasebinding/controller.go#L284

File: internal/networkpolicy/networkpolicy.go [hunk 2]
```diff
@@ -169,6 +192,168 @@ func makeIngressRules(params ComponentPolicyParams) []any {
...
+			"ports": l7Ports,
+			"rules": map[string]any{"http": []any{
+				map[string]any{},
+			}},
+		})
+	}
+	if len(l4Ports) > 0 {
+		toPorts = append(toPorts, map[string]any{"ports": l4Ports})
+	}
+	return toPorts
```
[the remark this answers is not in the document]
Comment (akila-i) [reply internal/networkpolicy/networkpolicy.go:342]: User needs to declare two endpoints on the same port — one HTTP (L7) and one TCP (L4) for this case to occur, which is unlikely. Adding validations at CEL expressions level when defining endpoints for workloads should be the ideal fix. Since it is not restricted at the moment, we'll keep this behavior as it is for the CNP as well.

**Review by NomadXD — APPROVED** [review]
